File: ImplementationTests/Runners/libWexpr.js/WexprValidate.js
#!/usr/bin/env node
//
// WexprValidate.js
// Simple program designed to be compatible with libWexpr's test runner
//
// Usage: [node] ./WexprValidate.js [wexprfilename]
// Returns 0 on success, or 1 if invalid wexpr
//

// put libWexpr.js in the same dir as this file
require('libWexpr.js');

var fs = require('fs')

// For now, the only argument we take is the input file
// argument one is our program
inputFilePath = process.argv[2];

fs.readFile (inputFilePath, 'utf8', function(err, data) {
	if (err) {
		process.exit(1);
	}

	try {
		res = libWexpr.decode(data);
	} catch (err) {
		console.log ("exception");
		process.exit(1); // failure due to exception
	}
	
	// res[0] is data, res[1] is success
	if (res[1] != null) {
		console.log ("Error: " + res[1]);
		process.exit(1); // failure
	}

	// success
	process.exit();
})

File: ImplementationTests/Runners/libWexpr.lua/WexprValidate.lua
#!/usr/bin/env lua

package.path ='?.lua;' .. package.path
libWexpr = require("libWexpr") -- put libWexpr.lua in the same dir as this file

path = arg[1]

local file = io.open(path, "rb")
local content = file:read("*a")
file:close()

local res,err = libWexpr:decode(content)

if res == nil and err ~= nil then
	print("Error: " .. err)
	os.exit(1)
end
